conan/internal/cache/cache.py

```diff
@@ -187,9 +188,16 @@ def assign_prev(self, layout: PackageLayout):
             self._db.create_package(relpath, pref, build_id)
         except ConanReferenceAlreadyExistsInDB:
             # TODO: Optimize this into 1 single UPSERT operation
-            # This was exported before, making it latest again, update timestamp
+            # There was a previous package folder for this same package reference (and prev)
             pkg_layout = self.get_package_layout(pref)
+            # We remove the old one and move the new one to the path of the previous one
+            # this can be necessary in case of new metadata or build-folder because of "build_id()"
             pkg_layout.remove()
+            shutil.move(layout.base_folder, pkg_layout.base_folder)  # clean unused temporary build
+            layout._base_folder = pkg_layout.base_folder  # reuse existing one
+            # TODO: The relpath would be the same as the previous one, it shouldn't be ncessary to
+            #  update it, the update_package_timestamp() can be simplified and path dropped
+            relpath = os.path.relpath(layout.base_folder, self._base_folder)
```

Comment:
I have little insight into why this is needed again given the conext of the comment above, I'll trust you on this one but maybe I just need to look deeper into this bit of code functionality